tests/Query/ConfigLoader.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace DoctrineExtensions\Tests\Query;
+
+use Doctrine\ORM\Configuration;
+use Symfony\Component\Yaml\Parser;
+
+class ConfigLoader
+{
+    const MYSQL    = 'mysql';
+    const ORACLE   = 'oracle';
+    const POSTGRES = 'postgres';
+    const SQLITE   = 'sqlite';
+
+    /**
+     * @param Configuration $configuration
+     * @param string        $database
+     */
+    public static function load(Configuration $configuration, $database)
+    {
+        $parser = new Parser();
+        // Load the corresponding config file.
+        $parsed = $parser->parse( file_get_contents( realpath( __DIR__ . '/../../config/' . $database . '.yml' ) ) );
+
+        // Make sure it has the correct structure.
+        if (array_key_exists( 'doctrine', $parsed)) {
+            if (array_key_exists('orm', $parsed['doctrine'])) {
+                if (array_key_exists('dql', $parsed['doctrine']['orm'])) {
+                    $parsed = $parsed['doctrine']['orm']['dql'];
+                } else {
+                    return;
+                }
+            } else {
+                return;
+            }
+        } else {
+            return;
+        }
```

Comment:
Without testing the array keys then this will error out, which should lead to understanding what went wrong. The way it's currently written then the test would just fail saying they don't know about the function being tested, which is probably the most difficult to debug. If I throw an exception with explicit instructions on what went wrong that should be the easiest to understand and fix. Thoughts?